src/m365/spe/commands/containertype/containertype-list.ts

```diff
@@ -0,0 +1,68 @@
+import { Logger } from '../../../../cli/Logger.js';
+import config from '../../../../config.js';
+import request, { CliRequestOptions } from '../../../../request.js';
+import { ClientSvcResponse, ClientSvcResponseContents, FormDigestInfo, spo } from '../../../../utils/spo.js';
+import SpoCommand from '../../../base/SpoCommand.js';
+import commands from '../../commands.js';
+import { ContainerTypeProperties } from '../../ContainerTypeProperties.js';
+
+class SpeContainertypeListCommand extends SpoCommand {
+  private allContainerTypes?: ContainerTypeProperties[];
+
+  public get name(): string {
+    return commands.CONTAINERTYPE_LIST;
+  }
+
+  public get description(): string {
+    return 'Lists all Container Types';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['ContainerTypeId', 'ContainerTypeName', 'OwningApplicationId'];
+  }
+  public async commandAction(logger: Logger): Promise<void> {
+    try {
+      const spoAdminUrl = await spo.getSpoAdminUrl(logger, this.debug);
+
+      if (this.verbose) {
+        await logger.logToStderr(`Retrieving list of Container types...`);
+      }
+
+      this.allContainerTypes = [];
+
+      await this.getAllContainerTypes(spoAdminUrl, logger);
+
+      await logger.log(this.allContainerTypes);
+    }
+    catch (err: any) {
+      this.handleRejectedPromise(err);
+    }
+  }
+
+  private async getAllContainerTypes(spoAdminUrl: string, logger: Logger): Promise<void> {
+    const formDigest: FormDigestInfo | undefined = undefined;
+    const formDigestInfo: FormDigestInfo = await spo.ensureFormDigest(spoAdminUrl, logger, formDigest, this.debug);
+
+    const requestOptions: CliRequestOptions = {
+      url: `${spoAdminUrl}/_vti_bin/client.svc/ProcessQuery`,
+      headers: {
+        'X-RequestDigest': formDigestInfo.FormDigestValue
+      },
+      data: `<Request AddExpandoFieldTypeSuffix="true" SchemaVersion="15.0.0.0" LibraryVersion="16.0.0.0" ApplicationName="${config.applicationName}" xmlns="http://schemas.microsoft.com/sharepoint/clientquery/2009"><Actions><ObjectPath Id="46" ObjectPathId="45" /><Method Name="GetSPOContainerTypes" Id="47" ObjectPathId="45"><Parameters><Parameter Type="Enum">1</Parameter></Parameters></Method></Actions><ObjectPaths><Constructor Id="45" TypeId="{268004ae-ef6b-4e9b-8425-127220d84719}" /></ObjectPaths></Request>`
+    };
+
+    const res: string = await request.post(requestOptions);
+    const json: ClientSvcResponse = JSON.parse(res);
+    const response: ClientSvcResponseContents = json[0];
+
+    if (response.ErrorInfo) {
+      throw (response.ErrorInfo.ErrorMessage);
+    }
+    else {
+      const containerTypes: ContainerTypeProperties[] = json[json.length - 1];
+      this.allContainerTypes!.push(...containerTypes);
+    }
```

Comment:
so we may correct a few things here:
1. we don't need the `else` statement because in the `if` we throw so it is not possible to go further as the `throw` will step out of the method in this case we will not go further. That is why we may just not use the `else` block statement and increase readabilty 
2. lets return the response instead of writing it to a local variable. Returning values instead of assigning some variable outside of the method increase reusability of the method
3. the `throw` does not require brackets

```suggestion
    if (response.ErrorInfo) {
      throw response.ErrorInfo.ErrorMessage;
    }

    const containerTypes: ContainerTypeProperties[] = json[json.length - 1];
    return containerTypes;
```

Reply:
That shows how less I write code, thanks for the explanation and code changes:)